### DISCUSSION TITLE
How to prevent prisma migrate to create migration to delete my GIS index?

### DISCUSSION THREAD

**Discussion by PavelNartov** [prisma/prisma#17719] [Q&A]
How to prevent `prisma migrate dev` to create new migration with:

```
-- AlterTable
ALTER TABLE "channels" ALTER COLUMN "textsearchable_index_col" DROP DEFAULT;
```

When I created index by previous migration:

```
-- AlterTable
ALTER TABLE "channels" ADD COLUMN     "textsearchable_index_col" tsvector
  GENERATED ALWAYS AS (to_tsvector('english', coalesce(domain, '') || ' ' || coalesce(title, ''))) STORED;

-- CreateIndex
CREATE INDEX "channels_textsearchable_index_col_idx" ON "channels" USING GIN ("textsearchable_index_col");
```

I have this schema.prisma file:

```
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["fullTextSearch", "extendedIndexes"]
}
...
model Channel {
...
textsearchable_index_col Unsupported("tsvector")?
@@index([textsearchableIndexCol], type: Gin, name: "channels_textsearchable_index_col_idx")
}
```

And on each `prisma migrate dev` it creates new migration that trying to drop my index.
How to prevent this?

**Comment by nurul3101** [comment]
Hey @PavelNartov 👋 
Which version of Prisma are you using?

I believe you are running into this issue:
- https://github.com/prisma/prisma/issues/12914

[Edit: I see you already commented on the GitHub Issue: 👍 ]

**Comment by PavelNartov** [comment]
Hi @nurul3101 ,

Yes, I think I am falling in that issue. I've even ask my question inside that issue.

I use prisma version **4.7.1** with "**extendedIndexes**" and "**fullTextSearch**" previewFeatures.

**Comment by janpio** [comment]
Does this really drop the index? Doesn't it look like it is modifying the column `channels` as it does not understand the "default"? Which in this case I think is a generated column?

**Reply by PavelNartov** [reply]
I've fixed this with command `npx prisms db pull` - it adds `dbgenerated` (with all my `GENERATED ALWAYS AS` command for field.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
